File: python/剑指offer/面试题09. 用两个栈实现队列.py
class CQueue:

    def __init__(self):
        self.input = []
        self.output = []

    def appendTail(self, value: int) -> None:
        self.input.append(value)

    def deleteHead(self) -> int:
        if not self.input and not self.output:
            return -1
        if self.output:
            return self.output.pop()
        while self.input:
            self.output.append(self.input.pop())
        return self.output.pop()

File: go/剑指offer/面试题09. 用两个栈实现队列.go
package main

type CQueue struct {
	input  []int
	output []int
}

func Constructor() CQueue {
	return CQueue{
		input:  make([]int, 0),
		output: make([]int, 0),
	}
}

func (this *CQueue) AppendTail(value int) {
	this.input = append(this.input, value)
}

func (this *CQueue) DeleteHead() int {
	if len(this.input) == 0 && len(this.output) == 0 {
		return -1
	}
	if len(this.output) != 0 {
		x := this.output[len(this.output)-1]
		this.output = this.output[:len(this.output)-1]
		return x
	}
	for len(this.input) != 0 {
		x := this.input[len(this.input)-1]
		this.input = this.input[:len(this.input)-1]
		this.output = append(this.output, x)

	}
	x := this.output[len(this.output)-1]
	this.output = this.output[:len(this.output)-1]
	return x
}

/**
 * Your CQueue object will be instantiated and called as such:
 * obj := Constructor();
 * obj.AppendTail(value);
 * param_2 := obj.DeleteHead();
 */
